index.js

```diff
@@ -15,23 +16,50 @@ const map = {
 const regex = /<pre><code class="(.*)?">([\s\S]*?)<\/code><\/pre>/igm;
 const captionRegex = /<p><code>(.*?)\s(.*?)\n([\s\S]*)<\/code><\/p>/igm;
 
-const baseDir = hexo.base_dir;
 const rootPath = hexo.config.root || '/';
-const prismDir = path.join(baseDir, 'node_modules', 'prismjs');
-const prismThemeDir = path.join(prismDir, 'themes');
+
+const themeRegex = /^prism-(.*).css$/;
+const prismDir = require.resolve('prismjs/package.json').replace('package.json', '');
+
+const prismThemeDir = dirResolve('prismjs/themes');
+const extraThemeDir = dirResolve('prism-themes/themes');
+function toThemeMap(basePath, filename) {
+  const matches = filename.match(themeRegex);
+  if (!matches)
+    return;
+
+  return {
+    name: matches[1],
+    filename,
+    path: path.join(basePath, filename)
+  };
+}
+const standardThemes = fs.listDirSync(prismThemeDir).map(themeFileName => toThemeMap(prismThemeDir, themeFileName));
+const extraThemes = fs.listDirSync(extraThemeDir).map(themeFileName => toThemeMap(extraThemeDir, themeFileName));
+const themes = standardThemes.concat(extraThemes).filter(Boolean);
```

Comment:
Is the filter function here necessary? AFAIK it will ignore falsy value, but it seems like the previous `toThemeMap()` function always returns an object with values in it.